test/algorithms/test_partial_equivalence.cpp

```diff
@@ -0,0 +1,420 @@
+  c1.setLogicalQubitGarbage(5);
+  c1.setLogicalQubitAncillary(3);
+  c1.setLogicalQubitGarbage(3);
+  c1.setLogicalQubitAncillary(4);
+  c1.setLogicalQubitGarbage(4);
+  EXPECT_TRUE(dd::partialEquivalenceCheck(c1, c2, dd));
+}
+
+void partialEquivalencCheckingBenchmarks(
+    const std::unique_ptr<dd::Package<dd::DDPackageConfig>>& dd,
+    const size_t minN, const size_t maxN, const size_t reps,
+    const bool addAncilla) {
+  for (size_t n = minN; n < maxN; n++) {
+    std::chrono::microseconds totalTime{0};
+    std::uint16_t totalGates{0};
+    for (size_t k = 0; k < reps; k++) {
+      dd::Qubit d{0};
+      if (addAncilla) {
+        d = static_cast<dd::Qubit>(rand()) % static_cast<dd::Qubit>(n - 1) + 1;
+      } else {
+        d = static_cast<dd::Qubit>(n);
+      }
+      dd::Qubit m{0};
+      if (d == 1) {
+        m = 1;
+      } else {
+        m = static_cast<dd::Qubit>(rand()) % static_cast<dd::Qubit>(d - 1) + 1;
+      }
+      const auto [c1, c2] = dd::generateRandomBenchmark(n, d, m);
+
+      const auto start = std::chrono::high_resolution_clock::now();
+      const bool result = dd::partialEquivalenceCheck(c1, c2, dd);
+      // Get ending timepoint
+      const auto stop = std::chrono::high_resolution_clock::now();
+      const auto duration =
+          std::chrono::duration_cast<std::chrono::microseconds>(stop - start);
+
+      EXPECT_TRUE(result);
+
+      // std::cout << "\nnumber of qubits = " << n << "; data qubits = " << d
+      //           << "; measured qubits = " << m
+      //           << "; number of gates = " << c2.size() << "\n";
+      // std::cout << "time: " << static_cast<double>(duration.count()) /
+      // 1000000.
+      //           << " seconds\n";
+      totalTime += duration;
+      totalGates += static_cast<std::uint16_t>(c2.size());
+    }
+    std::cout << "\nnumber of qubits = " << n << "; number of reps = " << reps
+              << "; average time = "
+              << (static_cast<double>(totalTime.count()) /
+                  static_cast<double>(reps) / 1000000.)
+              << " seconds; average number of gates = "
+              << (static_cast<double>(totalGates) / static_cast<double>(reps))
+              << "\n";
+  }
+}
+
+TEST(PartialEquivalenceTest, Benchmark) {
+  const auto dd = std::make_unique<dd::Package<>>(20);
+  srand(55);
+  const size_t minN = 2;
+  const size_t maxN = 8;
+  const size_t reps = 10;
+  std::cout << "Partial equivalence check\n";
+  partialEquivalencCheckingBenchmarks(dd, minN, maxN, reps, true);
+}
+
+TEST(PartialEquivalenceTest, ZeroAncillaBenchmark) {
+  const auto dd = std::make_unique<dd::Package<>>(30);
+  const size_t minN = 3;
+  const size_t maxN = 15;
+  const size_t reps = 10;
+  std::cout << "Zero-ancilla partial equivalence check\n";
+  partialEquivalencCheckingBenchmarks(dd, minN, maxN, reps, false);
+}
+
+TEST(PartialEquivalenceTest, InvalidInput) {
+  const auto dd = std::make_unique<dd::Package<>>(30);
```

Comment:
```suggestion
  const auto dd = std::make_unique<dd::Package<>>(4);
```
No need to allocate a package larger than necessary.